File: app/src/main/java/com/marked/pixsee/ui/chat/custom/ImageHolder.java
package com.marked.pixsee.ui.chat.custom;

import android.content.Context;
import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.facebook.common.util.UriUtil;
import com.facebook.drawee.view.SimpleDraweeView;
import com.marked.pixsee.R;
import com.marked.pixsee.data.message.Message;
import com.marked.pixsee.ui.chat.data.MessageConstants;

/**
 * Created by Tudor Pop on 04-Dec-15.
 */
class ImageHolder extends RecyclerView.ViewHolder implements Bindable {
	private Context context;

	private SimpleDraweeView mImage;

	ImageHolder(View itemView, Context context) {
		super(itemView);
		this.context = context;
		this.mImage = (SimpleDraweeView) itemView.findViewById(R.id.imageNetwork);
	}

	@Override
	public void bindMessage(final Message message, final ChatAdapter.ChatInteraction chatInteraction) {
		itemView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				chatInteraction.chatClicked(v, message, getAdapterPosition());
			}
		});
		StringBuilder path = new StringBuilder(message.getData().get(MessageConstants.DATA_BODY));
		if (message.getMessageType() == MessageConstants.MessageType.ME_IMAGE)
			path.insert(0, UriUtil.LOCAL_FILE_SCHEME+"://");
		final Uri url = UriUtil.parseUriOrNull(path.toString());

		mImage.setImageURI(url, context);
		mImage.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				chatInteraction.imageClicked(v,url);
			}
		});
	}
}

File: app/src/main/java/com/marked/vifo/ui/adapter/viewholders/ImageHolder.kt
package com.marked.vifo.ui.adapter.viewholders;

import android.content.Context
import android.net.Uri
import android.support.v7.widget.RecyclerView
import android.view.View
import com.facebook.drawee.view.SimpleDraweeView
import com.marked.vifo.R
import com.marked.vifo.extra.MessageConstants
import com.marked.vifo.model.message.Message

/**
 * Created by Tudor Pop on 04-Dec-15.
 */
class ImageHolder(itemView: View,val context: Context) : RecyclerView.ViewHolder(itemView) {
	private val mImage = itemView.findViewById(R.id.imageNetwork) as SimpleDraweeView

	fun bindMessage(message: Message) {
		val url = Uri.parse(message.data.get(MessageConstants.DATA_BODY));
		mImage.setImageURI(url,context);
	}

}
